packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3419,17 +3419,19 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
   /// until it finds a [RenderObjectElement].
   RenderObject? get renderObject {
     RenderObject? result;
-    void visit(Element element) {
-      assert(result == null); // this verifies that there's only one child
-      if (element._lifecycleState == _ElementLifecycle.defunct) {
-        return;
-      } else if (element is RenderObjectElement) {
-        result = element.renderObject;
+    final Queue<Element> elements = Queue<Element>();
+    elements.addFirst(this);
+    while (elements.isNotEmpty) {
+      final Element current = elements.removeFirst();
+      if (current._lifecycleState == _ElementLifecycle.defunct) {
+        break;
```

Comment:
Could the code be further simplified by eliminating the `result` variable altogether and directly returning null here and `current.renderObject` below?

Reply:
Yep, done.